keps/sig-storage/20190129-csi-volume-resizing.md

```diff
@@ -0,0 +1,167 @@
+---
+title: Support for CSI volume resizing
+authors:
+  - "@gnufied
+owning-sig: sig-storage
+participating-sigs:
+  - sig-storage
+reviewers:
+  - @saad-ali
+  - @jsafrane
+approvers:
+  - @saad-ali
+  - @childsb
+creation-date: 2019-01-29
+last-updated: 2019-01-29
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Support for CSI volume resizing
+
+## Table of Contents
+
+Table of Contents
+=================
+
+   * [Support for CSI volume resizing](#support-for-csi-volume-resizing)
+      * [Table of Contents](#table-of-contents)
+   * [Table of Contents](#table-of-contents-1)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+         * [Non-Goals](#non-goals)
+      * [Proposal](#proposal)
+         * [External resize controller](#external-resize-controller)
+         * [Expansion on Kubelet](#expansion-on-kubelet)
+             * [Offline volume resizing on kubelet:](#offline-volume-resizing-on-kubelet)
+             * [Online volume resizing on kubelet:](#online-volume-resizing-on-kubelet)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Test Plan](#test-plan)
+      * [Graduation Criteria](#graduation-criteria)
+      * [Implementation History](#implementation-history)
+
+
+## Summary
+
+To bring CSI volumes in feature parity with in-tree volumes we need to implement support for resizing of CSI volumes.
+
+## Motivation
+
+We recently implemented volume resizing support in CSI specs. This proposal implements this feature for Kubernetes.
+Any CSI volume plugin that implements necessary part of CSI specs will become resizable.
+
+### Goals
+
+To enable expansion of CSI volumes used by `PersistentVolumeClaim`s that support volume expansion as a plugin capability.
+
+### Non-Goals
+
+The expansion capability of a CSI plugin will not be validated by using CSI RPC call when user edits the PVC(i.e existing resize admission controller will not make CSI RPC call).
+The responsibility of
+actually enabling expansion for certains storageclasses still falls on Kubernetes admin.
+
+## Proposal
+
+The design of CSI volume resizing is made of two parts.
+
+
+#### External resize controller
+
+To support resizing of CSI volumes an external resize controller will monitor all PVCs.  If a PVC meets following critirea for resizing, it will be added to
```

Comment:
Does the internal resize controller need any changes to ignore CSI volumes?

Reply:
no - this was resolved couple of releases back when another sig-storage member wanted to externally resize volumes.